tools/walletextension/accountmanager/account_manager.go

```diff
@@ -211,18 +214,36 @@ func executeSubscribe(client *rpc.EncRPCClient, req *RPCRequest, _ *interface{})
 		return fmt.Errorf("could not call %s with params %v. Cause: %w", req.Method, req.Params, err)
 	}
 
+	// We listen for incoming messages on the subscription.
 	go func() {
 		for {
 			select {
-			case <-ch:
-				// TODO - #453 - Route subscription events back to frontend.
+			case receivedLog := <-ch:
+				jsonLog, err := json.Marshal(receivedLog)
+				if err != nil {
+					log.Error("could not marshal received log to JSON. Cause: %s", err)
+				}
+				err = readWriter.WriteResponse(jsonLog)
+				if err != nil {
+					log.Error("could not write the JSON log to the websocket. Cause: %s", err)
+				}
 			case err = <-subscription.Err():
-				// TODO - #453 - Route error back to frontend.
+				// An error on this channel means the subscription has ended, so we exit the loop.
+				readWriter.HandleError(err.Error())
+				return
 			}
 		}
 	}()
 
-	// TODO - #453 - Unsubscribe the subscription if the websocket is closed.
+	// We periodically check if the websocket is closed, and terminate the subscription.
+	go func() {
+		for {
+			if readWriter.IsClosed() {
+				subscription.Unsubscribe()
```

Comment:
I think you need to break out of the for loop here or you might leak the goroutine.

Reply:
Thanks, fixed :)